templates/resources/tls_subscription.md.tmpl

```diff
@@ -78,11 +78,11 @@ data "aws_route53_zone" "demo" {
 
 # Set up DNS record for managed DNS domain validation method
 resource "aws_route53_record" "domain_validation" {
-  name            = fastly_tls_subscription.example.managed_dns_challenge.record_name
-  type            = fastly_tls_subscription.example.managed_dns_challenge.record_type
+  name            = fastly_tls_subscription.example.managed_dns_challenges.record_name
+  type            = fastly_tls_subscription.example.managed_dns_challenges.record_type
   zone_id         = data.aws_route53_zone.demo.id
   allow_overwrite = true
-  records         = [fastly_tls_subscription.example.managed_dns_challenge.record_value]
+  records         = [fastly_tls_subscription.example.managed_dns_challenges.record_value]
```

Comment:
No longer works as it's now TypeSet rather than TypeMap

Reply:
I'd agree that `for_each` would make more sense here - especially as I expect the typical workflow would be to add a record for each DNS challenge in the set. However I think that snippet needs a couple of minor snippets for it to work! 🙂

Reply:
@bengesoff Thanks.

```tf
resource "aws_route53_record" "domain_validation" {
  for_each = { for domain in fastly_tls_subscription.example.managed_dns_challenges : domain.record_name => domain }
  name            = each.value.record_name
  type            = each.value.record_type
  zone_id         = data.aws_route53_zone.demo.id
  allow_overwrite = true
  records         = [each.value.record_value]
  ttl             = 60
}
```

maybe something like this would work?

Reply:
I'd imagine you could use something like the following although I haven't tried it
```hcl
resource "aws_route53_record" "domain_validation" {
  for_each        = fastly_tls_subscription.example.managed_dns_challenges
  name            = each.record_name
  type            = each.record_type
  zone_id         = data.aws_route53_zone.demo.id
  allow_overwrite = true
  records         = [each.record_value]
  ttl             = 60
}
```

Reply:
ok, I'll test

Reply:
Ok, just tested this. My second suggestion worked fine. Your suggestion gave me the below error:

```tf
| Error: Invalid "each" attribute
│
│   on fastly.tf line 105, in resource "aws_route53_record" "demo":
│  105:   name            = each.record_name
│
│ The "each" object does not have an attribute named "record_name". The supported attributes are each.key and each.value, the current key and value pair of the "for_each" attribute set.
```

so I'll go with the second one. Thanks

Reply:
Oh you're right, sorry 🤦🏻‍♂️ It was supposed to be `each.value.record_name` but that also doesn't work as `for_each` must be either a map or a set of strings! 

I've just tested it as well and there's a minor error with both:
```
╷
│ Error: Invalid for_each argument
│ 
│   on main.tf line 41, in resource "aws_route53_record" "domain_validation":
│   41:   for_each = { for domain in fastly_tls_subscription.example.managed_dns_challenges : domain.record_name => domain }
│     ├────────────────
│     │ fastly_tls_subscription.example.managed_dns_challenges is a set of object, known only after apply
│ 
│ The "for_each" value depends on resource attributes that cannot be determined until apply, so Terraform cannot predict how many instances will be created. To work around this, use the -target argument to first apply only the resources that the for_each depends
│ on.
╵
```

Unfortunately it has to be applied in two stages to get around this - do you think that's worth mentioning somewhere?